scripts/mop_workspace/mop_workspace.py

```diff
@@ -0,0 +1,474 @@
+import argparse
+import os
+
+from concurrent.futures import ThreadPoolExecutor
+from datetime import timedelta, datetime
+from firecloud import api as fapi
+from google.cloud import storage  # note: please use version 2.5.0
+from io import open
+from six import string_types
+from toolz.itertoolz import partition_all
+from tqdm import tqdm
+from time import sleep
+
+
+def _entity_paginator(namespace, workspace, etype, page_size=500,
+                      filter_terms=None, sort_direction="asc"):
+    """Pages through the get_entities_query endpoint to get all entities in
+       the workspace without crashing.
+    """
+
+    page = 1
+    all_entities = []
+    # Make initial request
+
+    def get_page():
+        r = fapi.get_entities_query(namespace, workspace, etype, page=page,
+                                    page_size=page_size, sort_direction=sort_direction,
+                                    filter_terms=filter_terms)
+        fapi._check_response_code(r, 200)
+        return r.json()
+
+    response_body = get_page()
+    # Get the total number of pages
+    total_pages = response_body['resultMetadata']['filteredPageCount']
+
+    # append the first set of results
+    entities = response_body['results']
+    all_entities.extend(entities)
+    # Now iterate over remaining pages to retrieve all the results
+    page = 2
```

Comment:
you could remove this line if you move the `page += 1` line to the first step in the `while` block.